qiskit/optimization/algorithms/slsqp_optimizer.py

```diff
@@ -0,0 +1,175 @@
+        continuous variables, and otherwise, returns a message explaining the incompatibility.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            Returns a string describing the incompatibility.
+        """
+        # check whether there are variables of type other than continuous
+        if len(problem.variables) > problem.get_num_continuous_vars():
+            return 'The SLSQP optimizer supports only continuous variables'
+
+        return ''
+
+    def solve(self, problem: QuadraticProgram) -> OptimizationResult:
+        """Tries to solves the given problem using the optimizer.
+
+        Runs the optimizer to try to solve the optimization problem.
+
+        Args:
+            problem: The problem to be solved.
+
+        Returns:
+            The result of the optimizer applied to the problem.
+
+        Raises:
+            QiskitOptimizationError: If the problem is incompatible with the optimizer.
+        """
+        # check compatibility and raise exception if incompatible
+        msg = self.get_compatibility_msg(problem)
+        if len(msg) > 0:
+            raise QiskitOptimizationError('Incompatible problem: {}'.format(msg))
+
+        # construct quadratic objective function
+        def _objective(x):
+            return problem.objective.sense.value * problem.objective.evaluate(x)
+
+        def _objective_gradient(x):
+            return problem.objective.sense.value * problem.objective.evaluate_gradient(x)
+
+        # initialize constraints list
+        slsqp_bounds = []
+        slsqp_eq_constraints = []
+        slsqp_ineq_constraints = []
+
+        # add lower/upper bound constraints
+        for variable in problem.variables:
+            lowerbound = variable.lowerbound
+            upperbound = variable.upperbound
+            slsqp_bounds.append((lowerbound, upperbound))
+
+        # pylint: disable=no-member
+        # add linear and quadratic constraints
+        for constraint in cast(List[Constraint], problem.linear_constraints) + \
+                cast(List[Constraint], problem.quadratic_constraints):
+            rhs = constraint.rhs
+            sense = constraint.sense
+
+            if sense == Constraint.Sense.EQ:
+                slsqp_eq_constraints += [lambda x, rhs=rhs, c=constraint: rhs - c.evaluate(x)]
+            elif sense == Constraint.Sense.LE:
+                slsqp_ineq_constraints += [lambda x, rhs=rhs, c=constraint: rhs - c.evaluate(x)]
+            elif sense == Constraint.Sense.GE:
+                slsqp_ineq_constraints += [lambda x, rhs=rhs, c=constraint: c.evaluate(x) - rhs]
+            else:
+                raise QiskitOptimizationError('Unsupported constraint type!')
+
+        fval_sol = INFINITY
+        x_sol = None    # type: Optional[np.array]
+
+        # Implementation of multi-start SLSQP optimizer
+        for trial in range(self._trials):
```

Comment:
@adekusar-drl could you please add the multi-start logic also to the COBYLA wrapper?

Reply:
@stefan-woerner Introduced `MultiStartOptimizer` that handles this multi start logic, both `SlsqpOptimizer` and `CobylaOptimizer` extend the new one.